scala/integration/intellij-bsp/src/org/jetbrains/plugins/scala/bsp/server/connection/ScalaConnectionDetailsProvider.scala

```diff
@@ -0,0 +1,127 @@
+package org.jetbrains.plugins.scala.bsp.server.connection
+
+import ch.epfl.scala.bsp4j.BspConnectionDetails
+import com.intellij.modcommand.ModCommand.error
+import com.intellij.openapi.components.{PersistentStateComponent, Service, State, Storage, StoragePathMacros}
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.util.JavaCoroutines
+import kotlin.coroutines.Continuation
+import org.jetbrains.bsp.protocol.utils.ParsersKt
+import org.jetbrains.plugins.bsp.extension.points.BuildToolId
+import org.jetbrains.plugins.bsp.server.connection.ConnectionDetailsProviderExtension
+import org.jetbrains.plugins.scala.bsp.config.ScalaPluginConstants.BUILD_TOOL_ID
+
+import java.lang
+import java.util.concurrent.TimeUnit
+import scala.jdk.CollectionConverters._
+
+class ScalaConnectionDetailsProvider extends ConnectionDetailsProviderExtension {
+  override def onFirstOpening(project: Project, projectPath: VirtualFile, continuation: Continuation[_ >: lang.Boolean]): AnyRef = {
+    //noinspection ApiStatus,UnstableApiUsage
+    JavaCoroutines.suspendJava[lang.Boolean](cont => {
+      saveProjectRootFile(project, projectPath)
+      val connectionFile = getConnectionFile(projectPath)
+      if (connectionFile.isDefined) {
+        cont.resume(true)
+      }
+
+      val hasGeneratedFile = generateSbtConnectionFile(projectPath).isDefined
+      cont.resume(hasGeneratedFile)
+    }, continuation)
```

Comment:
sanity check: have we confirmed that this is the correct way to call suspending functions from scala? that is - are the arguments (cont and continuation) used correctly? + is the AnyRef return method correct?

Reply:
We didn't find any documentation on that topic, but we found that it's used in intellij-scala and we mimicked the usage in our part.

Ref: https://github.com/JetBrains/intellij-scala/blob/idea241.x/scala/scala-impl/src/org/jetbrains/plugins/scala/startup/ProjectActivity.scala